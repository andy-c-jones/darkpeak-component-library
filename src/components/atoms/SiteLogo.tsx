type SiteLogoProps = {
  // The href of the site logo
  href: string;
  // The image source of the site logo
  src: string;
  // The alt text of the site logo
  alt: string;
};

// The site logo component
export const SiteLogo = ({ href, ...props }: SiteLogoProps) => {
  return (
    <a href={href}>
      <img {...props}></img>
    </a>
  );
};
