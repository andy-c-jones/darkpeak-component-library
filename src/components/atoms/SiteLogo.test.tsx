/**
 * @vitest-environment jsdom
 */
import { render, cleanup, screen } from '@testing-library/react';
import { describe, expect, it, beforeEach } from 'vitest';
import { SiteLogo } from './SiteLogo';

beforeEach(() => {
  cleanup();
});

describe('SiteLogo', () => {
  it('renders without error', () => {
    const { asFragment } = render(<SiteLogo href="test" src="test" alt="test" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders a link with the href passed in', () => {
    render(<SiteLogo href="myHref" src="someSrc" alt="someAltText" />);

    let link = screen.getByRole('link');

    expect(link.getAttribute('href')).toBe('myHref');
  });

  it('renders an image with the src passed in', () => {
    render(<SiteLogo href="someHref" src="mySrc" alt="someAltText" />);

    let image = screen.getByRole('img');

    expect(image.getAttribute('src')).toBe('mySrc');
  });

  it('renders an image with the alt passed in', () => {
    render(<SiteLogo href="someHref" src="someSrc" alt="myAltText" />);

    let image = screen.getByRole('img');

    expect(image.getAttribute('alt')).toBe('myAltText');
  });
});
