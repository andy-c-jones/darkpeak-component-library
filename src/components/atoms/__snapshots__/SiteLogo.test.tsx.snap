// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SiteLogo > renders without error 1`] = `
<DocumentFragment>
  <a
    href="test"
  >
    <img
      alt="test"
      src="test"
    />
  </a>
</DocumentFragment>
`;
